lib/bolt/transport/ssh/connection.rb

```diff
@@ -194,10 +194,10 @@ def execute(command, sudoable: false, **options)
           use_sudo = escalate && @target.options['run-as-command'].nil?
 
           command_str = inject_interpreter(options[:interpreter], command)
-
           if escalate
             if use_sudo
-              sudo_flags = ["sudo", "-S", "-u", run_as, "-p", Sudoable.sudo_prompt]
+              sudo_exec = target.options['sudo-executable'] || "sudo"
+              sudo_flags = [sudo_exec, "-S", "-u", run_as, "-p", Sudoable.sudo_prompt]
```

Comment:
I *think* we can add a test to `spec/integration/transport/ssh_spec.rb` along the lines of:
```
  context "with sudo as non-root", sudo: true do
    let(:config) {
      mk_config('host-key-check' => false, 'sudo-executable' => 'fake',
                'run-as' => user, user: bash_user, password: bash_password)
    }   
    let(:target) { make_target }

    it 'uses the correct executable' do
      expect_any_instance_of(Net::SSH::Connection::Channel).to receive(:exec)
        .with("fake -S -u bolt -p \\[sudo\\]\\ Bolt\\ needs\\ to\\ run\\ as\\ another\\ user,\\ password:\\  whoami", any_args)

      ssh.run_command(target, 'whoami')
    end 
  end 
```

This currently hangs for me though, I'm not quite sure how to handle the block param to `exec`.

Reply:
Ok, finally figured out what was hanging :P 
```
context "with sudo as non-root", sudo: true do
    let(:config) {
      mk_config('host-key-check' => false, 'sudo-executable' => 'fake',
                'run-as' => user, user: bash_user, password: bash_password)
    }   
    let(:target) { make_target }

    it 'uses the correct executable' do
      allow_any_instance_of(Net::SSH::Connection::Channel).to receive(:wait).and_return('')
      expect_any_instance_of(Net::SSH::Connection::Channel).to receive(:exec)
        .with("fake -S -u bolt -p \\[sudo\\]\\ Bolt\\ needs\\ to\\ run\\ as\\ another\\ user,\\ password:\\  whoami")

      ssh.run_command(target, 'whoami')
    end 
  end 
```